### PR TITLE
Clean up sprint 2: Complete test suite

### PR DESCRIPTION
**Pull request by paul891112**
Implement MockBuilders to simplify mocking objects

Add and match some test cases with current code for ControllerTest

Testing part should be complete after this PR

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs24/controller/Controller.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public String joinSession(@RequestBody JoinDTO joinData) {
         sessionService.validateSessionId(joinData.getSessionId());
 
         if (sessionService.hasStarted(joinData.getSessionId())) {
-            return null;
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "The game has already started, cannot join anymore");
```
Comment (C0DECYCLE) [inline src/main/java/ch/uzh/ifi/hase/soprafs24/controller/Controller.java:50]: I think this also need adaption on the client side.

**Review by C0DECYCLE — CHANGES_REQUESTED** [review]
LGTM, only the error thing, could you please also adapt the client.

File: src/main/java/ch/uzh/ifi/hase/soprafs24/controller/Controller.java (hunk 1 above)
Comment (paul891112) [reply src/main/java/ch/uzh/ifi/hase/soprafs24/controller/Controller.java:50]: I'll check that bybthe end of today.

The client should just display error message and stay in title screen right?

**Comment by C0DECYCLE** [comment]
@paul891112 or @Patertuck could you please create an issue for this under the clean up sprint and link it her and on the client. Thanks! **PS dont forget to write it into the merge commit message.**

**Comment by Patertuck** [comment]
> @paul891112 or @Patertuck could you please create an issue for this under the clean up sprint and link it her and on the client. Thanks! **PS dont forget to write it into the merge commit message.**

On it

**Review by C0DECYCLE — APPROVED** [review]
LGTM

**Comment by C0DECYCLE** [comment]
194: ...

**Pull request merged**